custom_components/tahoma/climate_st.py

```diff
@@ -0,0 +1,242 @@
+        """Return the list of supported features."""
+        return SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return MAP_HVAC_MODES[self.select_state(ST_DEROGATION_TYPE_STATE)]
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes."""
+        return [HVAC_MODE_AUTO, HVAC_MODE_HEAT]
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
+        return MAP_PRESET_MODES[
+            self.select_state(ST_HEATING_MODE_STATE)
+            if self.hvac_mode == HVAC_MODE_AUTO
+            else self.select_state(ST_DEROGATION_HEATING_MODE_STATE)
+        ]
+
+    @property
+    def preset_modes(self) -> Optional[List[str]]:
+        """Return a list of available preset modes."""
+        return [
+            PRESET_NONE,
+            PRESET_FREEZE,
+            PRESET_NIGHT,
+            PRESET_AWAY,
+            PRESET_HOME,
+        ]
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return 15.0
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return 26.0
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return (
+            self.select_state(PRESET_TEMPERATURES[self.preset_mode])
+            if self.hvac_mode == HVAC_MODE_AUTO
+            else self.select_state(CORE_DEROGATED_TARGET_TEMPERATURE_STATE)
+        )
+
+    def set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+
+        self.apply_action(
+            COMMAND_SET_DEROGATION, temperature, STATE_DEROGATION_FURTHER_NOTICE
+        )
+        self.apply_action(
+            COMMAND_SET_MODE_TEMPERATURE, STATE_PRESET_MANUAL, temperature
+        )
+        self.apply_action(COMMAND_REFRESH_STATE)
+
+    def set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        if hvac_mode == self.hvac_mode:
+            return
+        if hvac_mode == HVAC_MODE_AUTO:
+            self._saved_target_temp = self.target_temperature
+            self.apply_action(COMMAND_EXIT_DEROGATION)
+            self.apply_action(COMMAND_REFRESH_STATE)
+        elif hvac_mode == HVAC_MODE_HEAT:
+            self.set_preset_mode(PRESET_NONE)
+
+    def set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        if self.preset_mode == preset_mode:
+            return
+        modes = {
```

Comment:
Can be set as constant at the top of the module.

Reply:
it can, but as it's only used there, I thought it would be clearer. I'll move it.